Tests/test_death_handling.gd

```diff
@@ -50,15 +50,15 @@ func before_each():
 		PhaseManager.disconnect("on_combat_end", SceneController._combat_end_change_scene)
 
 	
-func _free_if_valid(node: Node):
+func _free_if_valid(node):
```

Comment:
missing typing, you removed it when it was already there; any reason ?

Reply:
If the node is invalid (as it was freed when the enemy died), it is no longer a node type. 

I don't think godot has option for multi-typing and even if it did, I'm not sure what type would even cover invalid node if any 🤔

Reply:
mmh I see the problem indeed;
could you put a comment so we don't forget about this exception ?

Reply:
Added